tests/test_runner.py

```diff
@@ -134,3 +134,11 @@ def test_experimental_all_models_register_automatic(testdir):
     """
     result = successful_test_run(testdir, num_tests=5, test_alembic=True)
     assert "test_all_models_register_on_metadata" in result
+
+
+def test_consistency_doesnt_roundtrip(testdir):
+    """Assert a up/down consistency fails if a migration cannot rountrip up -> down -> up."""
```

Comment:
Would we want to go up->down->up->down to leave the db in the down state?

Reply:
It shouldn't really matter, i dont think (unless i'm missing your point.

The test itself is verifying that your downgrade is putting the database in a similar enough state that your up works the 2nd time (which would generally catch typical up migrations with DDL changes).

But if you're just talking about cleanup-wise; in general in these tests, we leave the database in whatever state the test happens to leave it in. It'd be the job of the fixture infrastructure to either clean up (or in pytest-mock-resource's case, hand you a clean database the next time around).